### PR TITLE
Merge linkstateior to develop

### PR DESCRIPTION
**Pull request by MoLi7**
This PR will stay open till we complete updating `linkstateior`, including the integration of `stateior`, modifying disaggregagtion of summary level models, updating satellite, demand, LCIA, and build model functions.

### CODE REVIEW

**Comment by MoLi7** [comment]
@bl-young let's temporarily suspend validation check on state EEIO models in Actions workflow, otherwise the CI/CD test will never pass.

**Comment by bl-young** [comment]
@MoLi7 I'm able to build a complete GAEEIOv2.0-s-WAT now - great job!
I get a warning about `object 'InternationalTradeAdjustment' not found`. Would we expect that in a two region model?

**Comment by MoLi7** [comment]
> @MoLi7 I'm able to build a complete GAEEIOv2.0-s-WAT now - great job!

Awesome! Good to confirm you can do it, too.

> I get a warning about `object 'InternationalTradeAdjustment' not found`. Would we expect that in a two region model?

Yes, it should be included in a two-region model to support the creation of demand vectors. I'm working on solving it now.

**Comment by MoLi7** [comment]
@bl-young I'm puzzled by the failed workflow checks, this error message `Error: <callr_remote_error: Internal error, no task running and cannot select new task> in process 3634` is not making any sense to me. It looks like it's not GH's internal issue ([their services](https://www.githubstatus.com/) are all on). I wonder if it was caused by how `stateior` is required in DESCRIPTION, but I think it was all set up correctly.

**Comment by bl-young** [comment]
> @bl-young I'm puzzled by the failed workflow checks, this error message `Error: <callr_remote_error: Internal error, no task running and cannot select new task> in process 3634` is not making any sense to me. It looks like it's not GH's internal issue ([their services](https://www.githubstatus.com/) are all on). I wonder if it was caused by how `stateior` is required in DESCRIPTION, but I think it was all set up correctly.

Agreed, that was my guess given that's the main change here. @a-w-beck any thoughts on this error?

**Comment by MoLi7** [comment]
> > @bl-young I'm puzzled by the failed workflow checks, this error message `Error: <callr_remote_error: Internal error, no task running and cannot select new task> in process 3634` is not making any sense to me. It looks like it's not GH's internal issue ([their services](https://www.githubstatus.com/) are all on). I wonder if it was caused by how `stateior` is required in DESCRIPTION, but I think it was all set up correctly.
> 
> Agreed, that was my guess given that's the main change here. @a-w-beck any thoughts on this error?

Looking at [this example](https://r-pkgs.org/description.html#nonstandard-dependencies) of nonstandard dependencies, I guess it's probably not the most preferred to have `Remotes` section in DESCRIPTION. There will be a `v0.1.0` release of `stateior` soon, I'm hoping `Imports` a stable version will fix this error.

**Comment by bl-young** [comment]
@jvendries the disaggregation from the summary national files is mostly complete. The intersections are handled now such that first SoI is disaggregated, than RoUS. I am still getting some issues in the commodity mix [causing error here](https://github.com/USEPA/useeior/blob/linkstateior/R/IOFunctions.R#L73) that I have not yet been able to track down.

**Comment by bl-young** [comment]
@MoLi7 we are finding that the disaggregated model is not passing the commodity mix test (see [above](https://github.com/USEPA/useeior/pull/216#issuecomment-1043173663)). I believe this is due to the two-region tables themselves?

When running a non-disaggregated model ("GAEEIOv2.0-s-WAT"), `generateCommodityMixMatrix` returns a vector of 1's. I believe that `IndustryOutput` is coming directly from stateior.

When we disaggregate ("GAEEIOv2.0-79-WAT"), `IndustryOutput` instead is recalculated with `calculateIndustryCommodityOutput`. This is consistent with the national model which is also using Use Table column sums as the source of industry output. Based on looking at the Use table column sums in a two-region model without disaggregation, I believe `generateCommodityMixMatrix` would fail as well.

**Comment by MoLi7** [comment]
> @MoLi7 we are finding that the disaggregated model is not passing the commodity mix test (see [above](https://github.com/USEPA/useeior/pull/216#issuecomment-1043173663)). I believe this is due to the two-region tables themselves?
> 
> When running a non-disaggregated model ("GAEEIOv2.0-s-WAT"), `generateCommodityMixMatrix` returns a vector of 1's. I believe that `IndustryOutput` is coming directly from stateior.
> 
> When we disaggregate ("GAEEIOv2.0-79-WAT"), `IndustryOutput` instead is recalculated with `calculateIndustryCommodityOutput`. This is consistent with the national model which is also using Use Table column sums as the source of industry output. Based on looking at the Use table column sums in a two-region model without disaggregation, I believe `generateCommodityMixMatrix` would fail as well.

@bl-young could you remind me why, in `calculateIndustryCommodityOutput`, we decided to have `model$IndustryOutput <- colSums(model$UseTransactions) + colSums(model$UseValueAdded)` instead of `model$IndustryOutput <- rowSums(model$MakeTransactions)`?

I understand it wouldn't matter much at national level, because industry output summed from Make is approx. the same (or balanced) with that summed from Use. But at state level, there can be discrepancy between the two summed vectors due to truncation caused by the rounding to 1M $ in national IO tables.

In `stateior`, state industry and commodity output both are assured to converge to the row and column sums of Make table instead of Use. I wonder if we could modify `calculateIndustryCommodityOutput` to the following?
```
calculateIndustryCommodityOutput <- function(model) {
  model$IndustryOutput <- rowSums(model$MakeTransactions)
  model$CommodityOutput <- rowSums(model$UseTransactions) + rowSums(model$FinalDemand)
  return(model)
}
```

**Comment by bl-young** [comment]
> @bl-young could you remind me why, in `calculateIndustryCommodityOutput`, we decided to have `model$IndustryOutput <- colSums(model$UseTransactions) + colSums(model$UseValueAdded)` instead of `model$IndustryOutput <- rowSums(model$MakeTransactions)`?

I am trying to remember the reason, it may have had to do with disaggregation. I will look back to see if I can recall. @jvendries do you remember?

**Comment by MoLi7** [comment]
> > @bl-young could you remind me why, in `calculateIndustryCommodityOutput`, we decided to have `model$IndustryOutput <- colSums(model$UseTransactions) + colSums(model$UseValueAdded)` instead of `model$IndustryOutput <- rowSums(model$MakeTransactions)`?
> 
> I am trying to remember the reason, it may have had to do with disaggregation. I will look back to see if I can recall. @jvendries do you remember?

I vaguely remember we discussed about using Use as the core IO table to derive other IO data, namely industry and commodity output. Though the decision was to use Use, I don't think it rules out the feasibility of using Make, unless there was some critical factors that I missed.

**Comment by bl-young** [comment]
further discussion of this issue in #218

File: inst/extdata/modelspecs/GAEEIOv2.0-s-WAT.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ IOYear: 2012 # Year for IO data
 ModelRegionAcronyms: ["US-GA", "RoUS"]
 ModelType: "EEIO"
 IODataSource: "stateior"
+IODataVersion: "0.1.0"
```
Comment (MoLi7) [inline inst/extdata/modelspecs/GAEEIOv2.0-s-WAT.yml:8]: FYI, I added this new `param` to specify two-region IO data v`0.1.0` will be used for building the default GA model. For a disaggregated model e.g. `GAEEIOv2.0-79-WAT`, we can use for example v`0.2.0` to refer to the disaggregated two-region IO data.
@bl-young @jvendries @WesIngwersen

**Comment by MoLi7** [comment]
Safe to merge this one as well? @bl-young @jvendries

**Comment by bl-young** [comment]
> Safe to merge this one as well? @bl-young @jvendries

Isn't this PR mostly to handle disaggregation of the two region models and running two-region models in useeior? Has that been tested recently? I don't think this is ready and I didn't think it was going to be included in the current release.

**Comment by MoLi7** [comment]
> Isn't this PR mostly to handle disaggregation of the two region models and running two-region models in useeior? Has that been tested recently? I don't think this is ready and I didn't think it was going to be included in the current release.

Some changes in `linkstateior`, namely updates in `loadIOData.R`, have been pulled to `develop` and `updateIO`, so they  will be included in the current release as it's not easy to exclude them (revert the commits) at this point. Technically speaking, `linkstateior` is completed as all two-region IO tables can be successfully loaded and used. I understand state EEIO modeling is not done yet, so the current release will not mention it.

Apologies for the confusion, but I do hope to merge this PR as the first step (see my earlier [comment](https://github.com/USEPA/useeior/pull/221#issuecomment-1190694278)).

**Comment by bl-young** [comment]
ok i think that sounds fine. we should do some testing when all branches are combined before the relesae to make sure everything is working as expected (for the features we expect)

**Comment by MoLi7** [comment]
> ok i think that sounds fine. we should do some testing when all branches are combined before the relesae to make sure everything is working as expected (for the features we expect)

Certainly!

**Pull request merged**